rocAL/rocAL/include/meta_data/meta_data.h

```diff
@@ -81,150 +83,293 @@ typedef struct
+        _label_ids.clear();
+        _buffer_size.clear();
     }
     MetaDataBatch&  operator += (MetaDataBatch& other) override
     {
-        _label_id.insert(_label_id.end(), other.get_label_batch().begin(), other.get_label_batch().end());
+        _label_ids.insert(_label_ids.end(), other.get_labels_batch().begin(), other.get_labels_batch().end());
+        _info_batch += other.get_info_batch();
+        _metadata_dimensions.insert(other.get_metadata_dimensions_batch());
         return *this;
     }
     void resize(int batch_size) override
     {
-        _label_id.resize(batch_size);
+        _label_ids.resize(batch_size);
+        _info_batch.resize(batch_size);
+        _metadata_dimensions.resize(batch_size);
     }
     int size() override
     {
-        return _label_id.size();
+        return _label_ids.size();
     }
     std::shared_ptr<MetaDataBatch> clone() override
     {
         return std::make_shared<LabelBatch>(*this);
     }
-    explicit LabelBatch(std::vector<int>& labels)
+    explicit LabelBatch(std::vector<Labels>& labels)
     {
-        _label_id = std::move(labels);
+        _label_ids = std::move(labels);
     }
     LabelBatch() = default;
+    void copy_data(std::vector<void*> buffer) override
+    {
+        if(buffer.size() < 1)
+            THROW("The buffers are insufficient") // TODO -change
+        auto labels_buffer = (int *)buffer[0];
+        for (int i = 0; i < _label_ids.size(); i++) {
+            memcpy(labels_buffer, _label_ids[i].data(), _label_ids[i].size() * sizeof(int));
+            labels_buffer += _label_ids[i].size();
+        }
+    }
+    std::vector<size_t>& get_buffer_size() override
+    {
+        size_t size = 0;
+        for (auto label : _label_ids)
+            size += label.size();
+        _buffer_size.emplace_back(size * sizeof(int));
+        return _buffer_size;
+    }
+    std::vector<Labels>& get_labels_batch() { return _label_ids; }
+    protected:
+    std::vector<Labels> _label_ids = {};
+    std::vector<size_t> _buffer_size;
 };
 
-struct BoundingBoxBatch: public MetaDataBatch
+struct BoundingBoxBatch: public LabelBatch
 {
     void clear() override
     {
         _bb_cords.clear();
-        _bb_label_ids.clear();
-        _img_sizes.clear();
+        _label_ids.clear();
+        _info_batch.clear();
+        _metadata_dimensions.clear();
+        _buffer_size.clear();
     }
     MetaDataBatch&  operator += (MetaDataBatch& other) override
     {
         _bb_cords.insert(_bb_cords.end(), other.get_bb_cords_batch().begin(), other.get_bb_cords_batch().end());
-        _bb_label_ids.insert(_bb_label_ids.end(), other.get_bb_labels_batch().begin(), other.get_bb_labels_batch().end());
-        _img_sizes.insert(_img_sizes.end(), other.get_img_sizes_batch().begin(), other.get_img_sizes_batch().end());
+        _label_ids.insert(_label_ids.end(), other.get_labels_batch().begin(), other.get_labels_batch().end());
+        _info_batch += other.get_info_batch();
```

Comment:
Instead of overloading += operator we can add insert function similar to MetadaataDimensionsBatch